### ISSUE TITLE
How to set an alias that contains an assignment (an equal sign) ?

### ISSUE THREAD

**Issue by HTofi** [nushell/nushell#9308]
### Question

I had the following alias in `zsh` that I'd like to port into nushell: 
```
alias vide="WINIT_UNIX_BACKEND=x11 neovide --multigrid"
```

but it doesn't work in the standard way
```
alias vide = WINIT_UNIX_BACKEND=x11 neovide --multigrid
```

nor if I put quotations around the command. Enclosing the command with a `do`  environment:
```
alias vide = do { WINIT_UNIX_BACKEND=x11 neovide --multigrid }
```
works, but it doesn't allow giving any arguments to the command, for instance:
```
vide config.nu
```
will not open the file `config.nu`, but just an empty new file as if the argument wasn't there. So how can I fix this? 

Thanks in advance

**Comment by fdncred** [comment]
What version of nushell are you using? (you can use the `version` command to find out)

**Comment by fdncred** [comment]
@kubouch any tips?

I think another common one would be
```
alias cb = RUST_BACKTRACE=1 cargo build
```
then being about to do 
```
cb
```
or
```
cb --features=dataframe
```

**Comment by HTofi** [comment]
I'm using version 0.78.0

**Comment by fdncred** [comment]
> I'm using version 0.78.0

Maybe `old_alias` still works in 0.78.0?

**Comment by HTofi** [comment]
Nope, the command is not recognized

**Comment by fdncred** [comment]
> Nope, the command is not recognized

Sorry, it's a dash vs an underscore `old-alias`

**Comment by HTofi** [comment]
Yes, `old-alias` works:
```
old-alias vide = WINIT_UNIX_BACKEND=x11 neovide --multigrid
```

**Comment by HTofi** [comment]
but there is a problem when using tab completion. For instance, typing:
```
vide con
```
then pressing tab to complete into `config.nu` causes nushell to crash with the following message:
```
thread 'main' panicked at 'Can't create a Span whose end < start, start=18446744073709525464, end=8', /build/.cargo/registry/src/github.com-1ecc6299db9ec823/reedline-0.18.0/src/completion/base.rs:18:9
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

**Comment by fdncred** [comment]
This panic is one of the reasons why @kubouch rewrote aliases to work in a different way.

**Comment by kubouch** [comment]
Yeah, currently the env shorthands are not supported with aliases. I tried adding it but it ended up being quite complicated, I didn't finish it.

You could alias `with-env` instead: `alias foo = with-env { FOO: bar } { print $env.FOO }`.

**Comment by hamflx** [comment]
@kubouch If I use `with-env`, how do I pass the rest of the parameters?

**Comment by kubouch** [comment]
Not possible directly. You could define a custom command instead (possibly with the `--wrapped` flag if you want to pass any parameters to an external command):
```nushell
def --wrapped foo [...rest] {
    $env.FOO = 'bar'
    some-program $rest
}

foo --bar --baz
```

**Comment by WindSoilder** [comment]
Closing the issue because the question is answered.
